Detections/AzureDevOps/AzDOHistoricServiceConnectionAdds.yaml

```diff
@@ -0,0 +1,44 @@
+id: 5efb0cfd-063d-417a-803b-562eae5b0301
+name: Azure DevOps Service Conection Addition/Abuse - Historic Allowlist
+description: |
+  'This detection builds a allowlist of historic service connection use by Builds and Releases and compares to recent history, flagging growth of service connection use in non manually allowlisted, non historically allowlisted Build/Release runs.
+  This is to determine if someone is hijacking a build/release and adding many service connections in order to abuse or dump credentials from service connections.'
+severity: High
+queryFrequency: 2h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Execution
+  - Impact
+  - Escalation
+query: |
+
+  // Ignore Build/Releases with less/equal this number
+  let ServiceConnectionThreshold = 3;
+  // New Connections need to exhibit execution of more "new" connections than this number.
+  let NewConnectionThreshold = 1;
+  // List of Builds/Releases to ignore
+  let BypassDefIds = datatable(DefId:string, Type:string, ProjectName:string)
+  [
+  //"103", "Release", "ProjectA",
+  //"42", "Release", "ProjectB",
+  //"122", "Build", "ProjectB"
+  ];
+  let HistoricDefs = AzureDevOpsAuditing
+  | where TimeGenerated > ago(14d) and TimeGenerated < ago(6h)
```

Comment:
generally we specify these as parameters at the top - let start = 14d; let end = 6h; and then reference the parameter here.

Reply:
fixed with recentTimeframe